README.md

```diff
@@ -64,102 +70,122 @@ Please file an issue for missing permissions.
+(not *within* the root directory). This file allows you to set preprocessor flags
+that will be used when compiling the framework.
+
+We strongly recommend to start with a copy of the template config provided in this
+repository, `ISHPermissionKitAppConfiguration.xcconfig`. It includes a list of
+all supported flags, and you can easily specify which features you need by
+commenting or uncommenting the respective lines.
+
+You will have to use the same configuration file to build your app, else the
+category-specific symbols will not be available. In your project settings, you
+can select a configuration file for each target:
+
+![Weak-linking a framework in Xcode](assets/config_file.png)
+
+If you already use a configuration file, you can pick one and include the other
+in it. Ensure to always use `$(inherit)` when setting preprocessor macros.
+
 ### CocoaPods
 
-You can use CocoaPods to install ISHPermissionKit as a static library:
+You can use CocoaPods to install *ISHPermissionKit* as a static or dynamic library.
+Each permission category requires a separate (sub)pod. The following sample Podfile
+includes all available pods – you should pick only those that you are actually
+using in your app.
 
 ```ruby
 target 'MyApp' do
-  pod 'ISHPermissionKit'
+  use_frameworks! // remove this line if you want to link your pods statically
+  pod 'ISHPermissionKit/Motion'
+  pod 'ISHPermissionKit/Health'
+  pod 'ISHPermissionKit/Location'
+  pod 'ISHPermissionKit/Microphone'
+  pod 'ISHPermissionKit/PhotoLibrary'
+  pod 'ISHPermissionKit/Camera'
+  pod 'ISHPermissionKit/Notifications'
+  pod 'ISHPermissionKit/SocialAccounts'
+  pod 'ISHPermissionKit/Contacts'
+  pod 'ISHPermissionKit/Calendar'
+  pod 'ISHPermissionKit/Reminders'
+  pod 'ISHPermissionKit/Siri'
+  pod 'ISHPermissionKit/Speech'
+  pod 'ISHPermissionKit/Music'
 end
 ```
 
+[Providing a build configuration](#providing-a-build-configuration) manually is not
+required when you use CocoaPods.
+
 See the [official website](https://cocoapods.org/#get_started) to get started with
 CocoaPods.
 
-The default pod does not include HealthKit support. If you need HealthKit, you need
-to use the `ISHPermissionKit/Health` pod:
+### Required Frameworks
 
-```ruby
-target 'MyApp' do
-  pod 'ISHPermissionKit/Health'
-end
-```
+*ISHPermissionKit* uses system frameworks to accomplish its tasks. Most of
+them will be linked automatically unless you have disabled "Enable Modules"
+(`CLANG_ENABLE_MODULES`) and "Link Frameworks Automatically" 
+(`CLANG_MODULES_AUTOLINK`) in your app target's build settings.
 
-ISHPermissionKit can also be installed as a framework through CocoaPods:
+Unfortunately, some framework are not weakly linked automatically which
+will cause your app to crash at launch on older systems that don't support
+the respective framework. These frameworks must be explicitly linked in
+your app, and set to "Optional". Feel free to duplicate rdar://28008958.
```

Comment:
We should maybe publish the radar on open radar to make it easier for others to dupe it: https://openradar.appspot.com/search?query=28008958